pb-rs/src/types.rs

```diff
@@ -1476,6 +1485,22 @@ impl RpcService {
 
 pub type RpcGeneratorFunction = Box<dyn Fn(&RpcService, &mut dyn Write) -> Result<()>>;
 
+#[derive(Debug, Clone, Default)]
+pub struct Extensions {
+    pub from: i32,
+    /// Max number is 536,870,911 (2^29 - 1), as defined in the
+    /// protobuf docs
+    pub to: i32,
+}
+
+impl Extensions {
+    /// The max field number that can be used as an extension.
+    pub fn max() -> i32 {
+        536870911
+    }
+
+}
```

Comment:
To be clear this is related, but orthogonal functionality to 

```
extend X { ... }
```

This is the functionality for 

```
message A {
  ...
  extensions 500 to max
}

message B {
  ...
  extensions 39 to 339
}
```